app/controllers/PaymentScheduleController.scala

```diff
@@ -65,9 +66,13 @@ class PaymentScheduleController @Inject() (
         logErrorAndRouteToDefaultPageF(j)
 
       case j: Journey.AfterSelectedPaymentPlan =>
-        journeyService.updateHasCheckedPaymentPlan(j.journeyId)
-          .map(_ => Redirect(routes.BankDetailsController.typeOfAccount))
-
+        for {
+          _ <- journeyService.updateHasCheckedPaymentPlan(j.journeyId)
+          _ = j match {
```

Comment:
small thing here but could we audit first and then update mongo? With audit events it's best practice to stick the event as close as possible to the thing that triggered it. Also, in the very unlikely scenario that the BE update failed then we still have a chance of having the audit event there.

Reply:
Yeah sure thing